### ISSUE TITLE
How are the binaries built?

### ISSUE THREAD

**Issue by evancz** [kevva/elm-bin#21]
- I see there is a .tar.gz of linux things, but how are they made?
- Should I make these as part of the build process?
- Is there any reason to build 64-bit ones? I've heard from folks having trouble running them on 32-bit machines.
- Are the binaries for mac and windows taken from the installers I make?

**Comment by kevva** [comment]
* Currently, @passy is building them manually.
* Yes, ideally
* I've no idea, probably not
* I don't know, @passy?

**Comment by passy** [comment]
@evancz If you could provide linux binaries, that would be massively helpful.

Yes, I've built them myself so far and yup, they're 64bit only. Haven't personally come across any 32 bit machines for years, but if there's demand sure.

The Mac binaries are extracted from the pkg, Windows came from a PR. /cc @jvoigtlaender

**Comment by jvoigtlaender** [comment]
The Windows binaries I took from the official installer (https://github.com/kevva/elm-bin/pull/24).

**Comment by kevva** [comment]
> Yes, I've built them myself so far and yup, they're 64bit only. Haven't personally come across any 32 bit machines for years, but if there's demand sure.

@passy, me neither, but my experience from imagemin is that it seems there's always some edge case.

**Comment by dusky3** [comment]
Hi everyone, I would mega appreciate if I could install 32bit version via npm. I run Debian (Crunchbang) on my t42 and I spent the last 3 days trying to install the elm-platform. I have tried all the possible recommendations suggested by the others either here on github and on the other sites as well, but they all failed. I have read about what kind of problems do you encounter, but I really hope I will be able to run elm one day on my machine.

**Comment by jvoigtlaender** [comment]
@dusky3, I haven't seen you asking questions or reporting problems on https://github.com/elm-lang/elm-platform.

**Comment by dusky3** [comment]
@jvoigtlaender that's true, I haven't asked on https://github.com/elm-lang/elm-platform, beacuse I gave up and I don't want to go through that painfull process anymore. It's 2015 and IMHO no one should install any platform or framework by building it from source.

I demanded an installer here because I think that ELM is great and it should really have an npm installer for 32bit linux and as @passy claimed, if there's a demand he'll build it. @jvoigtlaender please help these guys here if necessary, I think it's worth it.

Anyway thank you everyone for all the hard work you've done so far.

**Comment by jvoigtlaender** [comment]
Sorry I'm not in the business of building Linux binaries.

But I feel some responsibility for the build-from-source script. Think of it like that: Some people do need to be able from source. For example developers that contribute to the Elm compiler etc., and people that build installers for different distributions.

So it would have been really good to see your failed attempts. When you say that all methods suggested on elm-platform don't work for you, I'm troubled. Any information from your three days trying this might have helped us to provide a fix for you (it could be a very simple thing, who knows), but also helped us to improve the process for other people.

So it's a lost opportunity for us that you suffered in silence instead of sharing. But of course you are under no obligation to open GitHub issues etc. It's just a pity.

**Comment by passy** [comment]
Totally agree with @jvoigtlaender. Unfortunately I'm tied up in too many other things and don't have a lot of time I can dedicate to this. And I have no idea how to cross-compile with GHC, so I won't be able to help out with the 32bit binaries, I'm afraid. :(

**Comment by evancz** [comment]
I'm planning to build *all* the binaries for 0.16 and I'll do 32-bit ones for Linux.

**Comment by passy** [comment]
@evancz That's excellent news! Thanks! :)

**Comment by dusky3** [comment]
Thank you @evancz, that's the news of the year! I will patiently wait for that.

@jvoigtlaender I know that the developers need to install from source, I was writing about regular users/programmers. Just to let you know - I just followed the newest instructions from here https://github.com/elm-lang/elm-platform/issues/63#issuecomment-145401034 , but again I had no luck. I will rather reply there.

**Comment by jvoigtlaender** [comment]
Thanks, I'll take a look.
